### DISCUSSION TITLE
Best process to import PDFs or converted PDF files for seamless recipes / app viewing

### DISCUSSION THREAD

**Discussion by delize** [TandoorRecipes/recipes#2689] [Q&A]
I have been using the external storage source for uploading individual PDF files and adding them as standalone recipes. This works Okay, but, not Fantastic. It is not exactly as seamless as I would like it to be. Namely, the lack of OCR (which I know is a huge development task) to Title, ingredients, or Instructions.  In addition, Untare (albeit not Recipes specific) cannot show the PDF files that are uploaded.

Is there any recommendations / guides that do not necessarily incorporate only Recipes' own functionality to get to a complete/proper ready to use import?

So far I have tried on a cookbook (that does not have headers or bookmarks):
* OCR file
* Convert to HTML
* Upload HTML Source into Recipes

Unfortunately this does not work out and provide issues with images (makes sense since the file path is local), instructions do not show up, ingredients do not show up.

Does anyone have a common method of exporting PDFs for import (without using the external files storage source)?

**Comment by smilerz** [comment]
Open-source OCR libraries are generally terrible and layouts too variable to automate. HTML isn't going to work either as the importer relies on structured data formats.

Manually typing (or copy /paste using your OS's built-in OCR) is going to be your best bet.  Note that you can paste a list of ingredients and it will convert it into amount/unit/good/note format.

**Comment by jdpedrie** [comment]
I wrote a script to convert images or recipes into a JSON Schema which is compatible with Tandoor using Chat GPT APIs.

https://github.com/jdpedrie/gpt-recipe/

**Reply by smilerz** [reply]
site not found

**Reply by smilerz** [reply]
@vabene1111 - we should put together a list of third party apps.  and put on front page?  documentation?  Tandoor itself?
There's now this, my menu generator, the kitchen display app, maybe others?

**Reply by jdpedrie** [reply]
Sorry, forgot to turn it public. Should work now.

**Reply by smilerz** [reply]
how sensitive is it to image quality?  taking pictures from books sometimes leads to weird glares, or warped text due to page spines

**Reply by jdpedrie** [reply]
I've only tried it on PDF files split into individual images so far. I don't have any printed cookbooks to test with, unfortunately.

**Reply by smilerz** [reply]
do you convert the PDFs into images first?  I'll give it a try at some point - I have about 10 cookbooks that need to be converted and I've been doing it manually for about a year now.

**Reply by jdpedrie** [reply]
Yeah. I used Automator to split the PDF into individual JPEGs. https://discussions.apple.com/thread/3311405
